docs/src/lib/device/wmma.md

```diff
@@ -0,0 +1,240 @@
+c_dev = CuArray(c)
+
+# Allocate space for result
+d_dev = similar(c_dev)
+
+# Matrix multiply-accumulate kernel (D = A * B + C)
+function kernel(a_dev, b_dev, c_dev, d_dev)
+    a_frag = llvm_wmma_load_a_col_m16n16k16_stride_f16(pointer(a_dev), 16)
+    b_frag = llvm_wmma_load_b_col_m16n16k16_stride_f16(pointer(b_dev), 16)
+    c_frag = llvm_wmma_load_c_col_m16n16k16_stride_f32(pointer(c_dev), 16)
+
+    d_frag = llvm_wmma_mma_col_col_m16n16k16_f32_f32(a_frag, b_frag, c_frag)
+
+    llvm_wmma_store_d_col_m16n16k16_stride_f32(pointer(d_dev), d_frag, 16)
+    return
+end
+
+@cuda threads=32 kernel(a_dev, b_dev, c_dev, d_dev)
+@test a * b + c ≈ Array(d_dev) rtol=0.01
+```
+
+## CUDA C-like API
+
+The main difference between the CUDA C-like API and the lower level wrappers, is that the former enforces several constraints when working with WMMA.
+For example, it ensures that the ``A`` fragment argument to the MMA instruction was obtained by a `load_a` call, and not by a `load_b` or `load_c`.
+Additionally, it makes sure that the data type and storage layout of the load/store operations and the MMA operation match.
+
+The CUDA C-like API heavily uses Julia's dispatch mechanism.
+As such, the method names are much shorter than the LLVM intrinsic wrappers, as most information is baked into the type of the arguments rather than the method name.
+
+
+Note that, in CUDA C++, the fragment is responsible for both the storage of intermediate results and the WMMA configuration.
+All CUDA C++ WMMA calls are function templates that take the resultant fragment as a by-reference argument.
+As a result, the type of this argument can be used during overload resolution to select the correct WMMA instruction to call.
+
+In contrast, the API in Julia separates the WMMA storage ([`WMMAFragment`](@ref)) and configuration ([`WMMAConfig`](@ref)).
+Instead of taking the resultant fragment by reference, the Julia functions just return it.
+This makes the dataflow clearer, but it also means that the type of that fragment cannot be used for selection of the correct WMMA instruction.
+Thus, there is still a limited amount of information that cannot be inferred from the argument types, but must nonetheless match for all WMMA operations, such as the overall shape of the MMA.
+This is accomplished by a separate "WMMA configuration" (see [`WMMAConfig`](@ref)) that you create once, and then give as an argument to all intrinsics.
+
+### Fragment
+```@docs
+CUDAnative.WMMAFragmentLayout
+CUDAnative.WMMARowMajor
+CUDAnative.WMMAColMajor
+CUDAnative.WMMAUnspecified
+CUDAnative.WMMAFragment
+```
+
+### WMMA configuration
+```@docs
+CUDAnative.WMMAConfig
+```
+
+### Load matrix
+```@docs
+CUDAnative.wmma_load_a
+CUDAnative.wmma_load_b
+CUDAnative.wmma_load_c
+```
+
+### Perform multiply-accumulate
+```@docs
+CUDAnative.wmma_mma
+```
+
+### Store matrix
+```@docs
+CUDAnative.wmma_store_d
+```
+
+### Fill fragment
+```@docs
+CUDAnative.wmma_fill_c
+```
+
+### Example
+
+```julia
```

Comment:
These examples are great to have. I am always a bit worries about making sure that the code doesn't go out of data and
one idea is to actually move them to https://github.com/JuliaGPU/CUDAnative.jl/tree/master/examples and the include them
here as 


    ````@eval
    using Markdown
    Markdown.parse("""
    ```julia
    $(read("../../examples/simple.jl", String))
    ```
    """)
    ````

iirc the examples folder is automatically CI tested.

Reply:
That's a great idea. I moved them to `examples/`, but now we need `CuArrays` as dependency for tests. I believe the changes to `Project.toml` and `Manifest.toml` should be sufficient for that?
If you prefer, I can also use `CuTestArray` like the other examples, but IMO it's more user friendly to use CuArrays.

Reply:
Ah that's an unfortunate dependency. We want to be able to develop and test CUDAnative without staggering changes in CuArray...
@maleadt will have to make this call

We could of course always cheat and include:

```
const CuArray = CuTestArray
```

Before loading the script (and somehow prevent CuArrays from being loaded...)

Alternatively add a test local package that aliases the CuArray name with a different UUID.

Reply:
Yeah, I'd rather not depend on CuArrays, but we can cheat with an alias like the existing examples do to keep it user friendly.